src/server/pfs/server/api_server.go

```diff
@@ -318,8 +320,50 @@ func (a *apiServer) PutFile(putFileServer pfs.API_PutFileServer) (retErr error)
 	return nil
 }
 
+type putFileHelperLog struct {
+	Service  string `json:"service,omitempty"`
+	Method   string `json:"method,omitempty"`
+	Request  string `json:"request,omitempty"`
+	Duration string `json:"duration,omitempty"`
+	Error    error  `json:"error,omitempty"`
+	FilePath string `json:"filepath,omitempty"`
+	ObjPath  string `json:"objpath,omitempty"`
+}
+
+func putFileLogHelper(request *pfs.PutFileRequest, err error, duration time.Duration, filePath string, objPath string) {
+	requestString := ""
+	if request != nil {
+		requestString = request.String()
+	}
+	l := &putFileHelperLog{
+		Service:  "pfs.API",
+		Method:   "putFileObjHelper",
+		Request:  requestString,
+		Duration: duration.String(),
+		Error:    err,
+		FilePath: filePath,
+		ObjPath:  objPath,
+	}
+	logJSON, err := json.Marshal(l)
```

Comment:
Actually you should just use the rpc logging method that's even better. `FilePath` and `ObjPath` fit nicely in that protobuf as the `File` field and the `URL` field. This way a recursive request will produce all the log messages that the individual requests would have produced.